src/main/src/main/java/org/geoserver/security/password/AbstractGeoserverPasswordEncoder.java

```diff
@@ -92,7 +92,8 @@ protected final PasswordEncoder getActualEncoder() {
 
     @Override
     public String encodePassword(String rawPass, Object salt) throws DataAccessException {
-        return doEncodePassword(getStringEncoder().encodePassword(rawPass, salt));
+        //        return doEncodePassword(getStringEncoder().encodePassword(rawPass, salt));
+        return doEncodePassword(getStringEncoder().encode(rawPass));
```

Comment:
The new Spring Security API uses random salts by default now instead of asking for a salt (incidentally all of our pre-existing encoders ignored the salt field anyways - they were also using random salts).

Reply:
A bit confused... if the salt is chosen at random how will a password be recognized next time the VM starts?

Reply:
The salt is stored alongside the password
https://security.stackexchange.com/a/66990